js/views/import_view.js

```diff
@@ -0,0 +1,153 @@
+/*
+ * vim: ts=4:sw=4:expandtab
+ */
+(function () {
+  'use strict';
+  window.Whisper = window.Whisper || {};
+
+  var State = {
+    IMPORTING: 1,
+    COMPLETE: 2
+  };
+
+  var IMPORT_STARTED = 'importStarted';
+  var IMPORT_COMPLETE = 'importComplete';
+  var IMPORT_LOCATION = 'importLocation';
+
+  Whisper.Import = {
+    isStarted: function() {
+      return Boolean(storage.get(IMPORT_STARTED));
+    },
+    isComplete: function() {
+      return Boolean(storage.get(IMPORT_COMPLETE));
+    },
+    isIncomplete: function() {
+      return this.isStarted() && !this.isComplete();
+    },
+    start: function() {
+      storage.put(IMPORT_STARTED, true);
+    },
+    complete: function() {
+      storage.put(IMPORT_COMPLETE, true);
+    },
+    saveLocation: function(location) {
+      storage.put(IMPORT_LOCATION, location);
+    },
+    reset: function() {
+      return Whisper.Backup.clearDatabase();
+    }
+  };
+
+  Whisper.ImportView = Whisper.View.extend({
+    templateName: 'app-migration-screen',
+    className: 'app-loading-screen',
+    events: {
+      'click .import': 'onImport',
+      'click .restart': 'onRestart',
+      'click .cancel': 'onCancel',
+    },
+    initialize: function() {
+      if (Whisper.Import.isIncomplete()) {
+        this.error = true;
+      }
+
+      this.render();
+      this.pending = Promise.resolve();
+    },
+    render_attributes: function() {
+      var message;
+      var importButton;
+      var hideProgress = true;
+      var restartButton;
+      var cancelButton;
+
+      if (this.error) {
+        return {
+          message: i18n('importError'),
+          hideProgress: true,
+          importButton: i18n('tryAgain'),
+        };
+      }
+
+      switch (this.state) {
+        case State.COMPLETE:
+          message = i18n('importComplete');
+          restartButton = i18n('restartSignal');
+          break;
+        case State.IMPORTING:
+          message = i18n('importing');
```

Comment:
Okay, this should be addressed. We don't erase the database or change to the 'in progress' view until the user has completed selecting the target directory.